utils/grpc/setup_server_test.go

```diff
@@ -0,0 +1,87 @@
+package grpc
+
+import (
+	"context"
+	"github.com/stretchr/testify/assert"
+	"github.com/tron-us/go-btfs-common/config"
+	hubpb "github.com/tron-us/go-btfs-common/protos/hub"
+	"github.com/tron-us/go-btfs-common/protos/shared"
+	sharedpb "github.com/tron-us/go-btfs-common/protos/shared"
+	"github.com/tron-us/go-common/v2/constant"
+	"github.com/tron-us/go-common/v2/log"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/connectivity"
+	"strings"
+	"testing"
+	"time"
+)
+type serverStruct struct {
+	hubpb.UnimplementedHubQueryServiceServer
+	sharedpb.UnimplementedRuntimeServiceServer
+}
+func TestSetupServer(t *testing.T) {
+
+	s := GrpcServer{}
+	config.InitDB()
+	address := "localhost:50030"
+	pgConMaps :=  map[string]string{"DB_URL_STATUS":config.DbStatusURL, "DB_URL_GUARD": config.DbGuardURL}
+	rdCon := config.RdURL
+	quit := make(chan struct{})
+
+	//create server
+	go func() {
+		for {
+			select {
+			case <-quit:
+				return
+			default:
+				s.GrpcServer(address, pgConMaps, rdCon, &serverStruct{})
```

Comment:
@uchetron Could you please help me explain the infinite for loop with default option in select? 

Correct me if I am wrong,  wont this will create infinite GrpcServers until the it receives quit signal.

Reply:
that's a little more like it, actually.